scripts/network.js

```diff
@@ -118,39 +143,56 @@ export class ExplorerNetwork extends Network {
         // In addition, always skip the coinbase transaction and in case the coinstake one
         // TODO: once v6.0 and shield stake is activated we might need to change this optimization
         for (const tx of block.txs.slice(skipCoinstake ? 2 : 1)) {
-            const r = await fetch(
-                `${this.strUrl}/api/v2/tx-specific/${tx.txid}`
-            );
-            if (!r.ok) throw new Error('failed');
-            const newTx = await r.json();
-            newTxs.push(newTx);
+            // Pull raw Tx from explorer - unless we used Nodes, then the full tx is already loaded
+            if (!fUseNodes) {
+                const r = await fetch(
+                    `${this.strUrl}/api/v2/tx-specific/${tx.txid}`
+                );
+                if (!r.ok) throw new Error('failed');
+                const newTx = await r.json();
+                newTxs.push(newTx);
+            } else {
+                newTxs.push(tx);
+            }
         }
         block.txs = newTxs;
         return block;
     }
 
     /**
-     * Fetch the block height of the current explorer
+     * Fetch the block height of the current explorer or fallback node
      * @returns {Promise<number>} - Block height
      */
     async getBlockCount() {
-        const { backend } = await (
-            await retryWrapper(fetchBlockbook, true, `/api/v2/api`)
-        ).json();
+        try {
+            // Attempt via Explorer first
+            const { backend } = await (
+                await retryWrapper(fetchBlockbook, true, `/api/v2/api`)
+            ).json();
 
-        return backend.blocks;
+            return backend.blocks;
+        } catch {
+            // Use Nodes as a fallback
+            return parseInt(await this.callRPC('/getblockcount', true));
+        }
     }
 
     /**
-     * Fetch the latest block hash of the current explorer
+     * Fetch the latest block hash of the current explorer or fallback node
      * @returns {Promise<string>} - Block hash
      */
     async getBestBlockHash() {
-        const { backend } = await (
-            await retryWrapper(fetchBlockbook, true, `/api/v2/api`)
-        ).json();
+        try {
+            // Attempt via Explorer first
+            const { backend } = await (
+                await retryWrapper(fetchBlockbook, true, `/api/v2/api`)
+            ).json();
 
-        return backend.bestBlockHash;
+            return backend.bestBlockHash;
+        } catch {
+            // Use Nodes as a fallback
```

Comment:
I don't like using nodes as a fallback for light RPC calls like `getBestBlockHash` , `sendTransaction` and `getBlockCount`: the code is more ugly and imo we should just make sure to have working explorers

Reply:
Imo we would need at least 2x (4) explorers to consider it a stable network, or we start relying more on Nodes instead, it is the only compromise to get MPW more stable.